sympy/solvers/ode/hypergeometric.py

```diff
@@ -50,6 +50,8 @@ def match_2nd_hypergeometric(eq, func):
             n, d = eq.as_numer_denom()
             eq = expand(n)
             r = collect(eq, [func.diff(x, 2), func.diff(x), func]).match(deq)
+            if not all(val.is_polynomial() for val in r.values()):
+                return []
```

Comment:
I'm not sure what the appropriate check should be here. Potentially the coefficients might be polynomial in the independent variable but not in some other symbols. Maybe that's okay though and the method should still match. Some more examples should be considered to see whether the solver matches and whether or not it is able to solve the ODE.

Reply:
I think `-x**(5/7)*(-416*x**(9/7)/9 - 2385*x**(5/7)/49 + 298*x/3)*f(x)/(196*(-x**(6/7) + x)**2*(x**(6/7) + x)**2) + Derivative(f(x), (x, 2))` should be `2nd order hypergeometric` but it instead gives `()`